services/tripService.js

```diff
@@ -37,6 +38,10 @@ export async function deleteTrip(id) {
   if (!validateId(id)) return { status: 400, error: "Invalid id" };
 
   await dbConnect();
+  const allDestinations = await getDestinationsByTripId(id);
+  allDestinations.map((destination) => deleteDestination(destination.id, id));
+
   const deletedTrip = await Trip.deleteOne({ _id: id });
-  return deletedTrip.deletedCount > 0 ? deletedTrip : { status: 404, error: `${id} not found` };
+  if (!deletedTrip.acknowledged) return { status: 404, error: `${id} not found` };
```

Comment:
This is a good way of checking whether an operation was successfull instead of checking the validity of ids everywhere. Your server will not crash if it is being called with an invalid id. If your logic to change the validity changes lateron, a lot of your calls might fail with your old data.